src/sonic-config-engine/minigraph.py

```diff
@@ -438,14 +446,20 @@ def parse_xml(filename, platform=None, port_config_file=None):
     results['VLAN_INTERFACE'] = vlan_intfs
     results['PORTCHANNEL_INTERFACE'] = pc_intfs
 
-    for port_name in port_speeds:
+    for port_name in port_speeds_default:
         # ignore port not in port_config.ini
         if not ports.has_key(port_name):
             continue
 
-        ports.setdefault(port_name, {})['speed'] = port_speeds[port_name]
-        if port_speeds[port_name] == '100000':
-            ports.setdefault(port_name, {})['fec'] = 'rs'
+        ports.setdefault(port_name, {})['speed'] = port_speeds_default[port_name]
+    for port_name in port_speed_png:
+        ports.setdefault(port_name, {})['speed'] = port_speed_png[port_name]
+    for port_name, port in ports.items():
+        if port.get('speed') == '100000':
```

Comment:
Is it guaranteed that 'speed' will be available for all ports?

Reply:
Reading code, seems not.

---
In reply to: [181201283](https://github.com/Azure/sonic-buildimage/pull/1592#discussion_r181201283) [](ancestors = 181201283)